ipfx/nwb_writer.py

```diff
@@ -0,0 +1,71 @@
+import pynwb
+from allensdk.core.nwb_data_set import NwbDataSet
+import ipfx.nwb_reader as nwb_reader
+
+
+class NwbWriter(object):
+
+    def __init__(self, nwb_file_name):
+        self.nwb_file_name = nwb_file_name
+
+    def add_spike_times(self, sweep_num, spike_times):
+        raise NotImplementedError
+
+    def save(self):
+        raise NotImplementedError
+
+
+class Nwb1Writer(NwbWriter):
+
+    def __init__(self, nwb_file_name):
+        NwbWriter.__init__(self, nwb_file_name)
+        self.nwbfile = NwbDataSet(self.nwb_file_name)
+
+    def add_spike_times(self, sweep_num, spike_times):
+        self.nwbfile.set_spike_times(sweep_num, spike_times)
+
+    def save(self):
+        pass
```

Comment:
Does nwb1writer not need to save?

Reply:
for nwb1, file operation is handled by the [NwbDataSet](https://github.com/AllenInstitute/AllenSDK/blob/a06728262f82bc92e0865311b7b82891edc11a06/allensdk/core/nwb_data_set.py#L258) class that uses context manager and closes file automatically on exit from context.